src/function/scalar/list/list_resize.cpp

```diff
@@ -0,0 +1,237 @@
+	D_ASSERT(args.data[1].GetType().id() == LogicalTypeId::UBIGINT);
+	if (result.GetType().id() == LogicalTypeId::SQLNULL) {
+		FlatVector::Validity(result).SetInvalid(0);
+		return;
+	}
+	auto child_type = ListType::GetChildType(args.data[0].GetType()).InternalType();
+	switch (child_type) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		TemplatedListResizeFunction<int8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT16:
+		TemplatedListResizeFunction<int16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT32:
+		TemplatedListResizeFunction<int32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT64:
+		TemplatedListResizeFunction<int64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT128:
+		TemplatedListResizeFunction<hugeint_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT8:
+		TemplatedListResizeFunction<uint8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT16:
+		TemplatedListResizeFunction<uint16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT32:
+		TemplatedListResizeFunction<uint32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT64:
+		TemplatedListResizeFunction<uint64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::FLOAT:
+		TemplatedListResizeFunction<float, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::DOUBLE:
+		TemplatedListResizeFunction<double, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::VARCHAR:
+		TemplatedListResizeFunction<string_t, StringCopyValue>(args, result);
+		break;
+	case PhysicalType::INTERVAL:
+		TemplatedListResizeFunction<interval_t, NestedCopyValue>(args, result);
+		break;
+	case PhysicalType::STRUCT:
+	case PhysicalType::LIST:
+		TemplatedListResizeFunction<int8_t, NestedCopyValue>(args, result);
+		break;
+	default:
+		throw NotImplementedException("This function has not been implemented for logical type %s",
+		                              TypeIdToString(child_type));
+	}
+}
+
+static unique_ptr<FunctionData> ListResizeBind(ClientContext &context, ScalarFunction &bound_function,
+                                               vector<unique_ptr<Expression>> &arguments) {
+	D_ASSERT(bound_function.arguments.size() == 2 || arguments.size() == 3);
+	bound_function.return_type = arguments[0]->return_type;
+	bound_function.arguments[1] = LogicalType::UBIGINT;
+	if (bound_function.arguments.size() == 3) {
+		if (arguments[0]->return_type == LogicalType::UNKNOWN) {
+			return make_uniq<VariableReturnBindData>(bound_function.return_type);
+		}
+		if (ListType::GetChildType(arguments[0]->return_type) != arguments[2]->return_type &&
+		    arguments[2]->return_type != LogicalTypeId::SQLNULL) {
+			throw InvalidInputException("Default value must be of the same type as the lists or NULL");
+		}
+	}
```

Comment:
Can we add the following tests? Maybe you already have them, and I missed them, but I think that we are missing a case here where the `LIST` does not have a child type (`arguments[0]->return_type == LogicalType::SQLNULL`). We need to set the return type to `SQLNULL` in that case, to trigger the following code in `ListResizeFunction`.
```cpp
	if (result.GetType().id() == LogicalTypeId::SQLNULL) {
		FlatVector::Validity(result).SetInvalid(0);
		return;
	}
```
```sql
query I
SELECT list_resize(NULL, 1, 1);
----
NULL

query I
SELECT list_resize(NULL, 2, NULL);
----
NULL

query I
SELECT list_resize(NULL, NULL, NULL);
----
NULL
```